src/tools/perf/lib/libperf.c

```diff
@@ -1796,9 +1864,27 @@ static ucs_status_t uct_perf_create_md(ucx_perf_context_t *perf)
 
     ucs_error("Cannot use "UCT_PERF_TEST_PARAMS_FMT,
               UCT_PERF_TEST_PARAMS_ARG(&perf->params));
+
+    if (!is_valid_device && !is_valid_tl) {
```

Comment:
maybe something like
if (!ucs_string_set_is_empty(devices_hint)) {
   // print available devices
} else if (!ucs_string_set_is_empty(tls_hint)) {
    // print available tls
} else {
    // print all devices and transprots
}